nntrainer/models/neuralnet.cpp

```diff
@@ -591,45 +613,57 @@ int NeuralNetwork::train_run() {
   auto &label = last_layer_node->getOutputGrad(0);
   auto &in = first_layer_node->getInput(0);
 
+  /// below constant is needed after changing
+  /// databuffer having train, valid, test -> train buffer, valid buffer, test
+  /// buffer After the cahgne, only data train is used inside a databuffer.
+  /// RUN_CONSTANT is a stub value to deal with the situation
+  auto RUN_CONSTANT = DatasetDataUsageType::DATA_TRAIN;
+
+  auto &[train_buffer, valid_buffer, test_buffer] = data_buffers;
+
+  if (train_buffer == nullptr) {
```

Comment:
This is outside the scope of this PR, but reduce to function to take a dataset and just run with it depending on its type.
This will simplify this function, and databuffer usage.

Reply:
Yup, while refactoring, I'll do in this manner :)

Reply:
when you do the refactoring, you may need to consider the thread conflict as well.